File: LeetCode/Easy/1299. Replace Elements with Greatest Element on Right Side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        
        int max = 0;
        
        for(int x = arr.length - 1; x >= 0; x--){
            
            if (x == arr.length - 1){
                max = arr[x];
            }
            else{
                int temp = max;
                if (arr[x] > max){
                    max = arr[x];
                }
                
                arr[x] = temp;
            }
        }
        
        arr[arr.length - 1] = -1;
        
        return arr;
    }
}

File: LeetCode/Python/Easy/1299. Replace Elements with Greatest Element on Right Side.py
"""
Given an array arr, replace every element in that array with the greatest element among the elements to its right, and replace the last element with -1.

After doing so, return the array.
"""

class Solution(object):
    def replaceElements(self, arr):
        val = 0
        for x in range(len(arr)):
            pos = len(arr) - x - 1
            if pos == len(arr) - 1:
                val = arr[pos]
                arr[pos] = -1
            else:
                temp = arr[pos]
                arr[pos] = val
                if temp > val:
                    val = temp

        return arr
        
